### PR TITLE
Features/civ 6375 settlement agreement

### PR DESCRIPTION
**Pull request by RosDiaf**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-6375

### Change description ###

**Scope:** The scope of this ticket to develop the screen "Terms of the agreement" when claimant has selected option "Sign a settlement agreement" for the formalise repayment.

**The Agreement will page will display information about the repayment plan.**. at top of the page and a check box to agree the plan.

(for example)
**The agreement**	Mrs. Jane Doe will repay £200 in instalments of £50 every month. The first instalment will be paid by 1 January 2023.
**Completion date**	1 May 2023

plus some other static text.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/routes/guards/signSettlmentAgreementGuard.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import {constructResponseUrlWithIdParams} from '../../common/utils/urlFormatter';
+import {Claim} from '../../common/models/claim';
+import {getCaseDataFromStore} from '../../modules/draft-store/draftStoreService';
+
+export class SignSettlmentAgreementGuard {
+  static apply(redirectUrl: string) {
+    return async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+      try {
+        const caseData: Claim = await getCaseDataFromStore(req.params.id);
+        if (caseData.isPartialAdmission() || caseData.isFullAdmission()) {
```
Comment (GorilaNaranja) [inline src/main/routes/guards/signSettlmentAgreementGuard.ts:11]: Do we need this first condition?
I mean, if we have `caseData.partialAdmission?.paymentIntention?.repaymentPlan` or `caseData.fullAdmission?.paymentIntention?.repaymentPlan` that means that is already true isPartialAdmission and isFullAdmission

**Review by laperla11 — APPROVED** [review]

**Comment by portizdev** [comment]
Please fill in the description of the PR so we don't have to go looking for the Jira ticket and have more or less an explanation of what it is about.

File: src/main/routes/guards/signSettlmentAgreementGuard.ts (hunk 1 above)
Comment (RosDiaf) [reply src/main/routes/guards/signSettlmentAgreementGuard.ts]: if (caseData.isPartialAdmission() || caseData.isFullAdmission()) removed

**Comment by RosDiaf** [comment]
**Scope:** The scope of this ticket to develop the screen "Terms of the agreement" when claimant has selected option "Sign a settlement agreement" for the formalise repayment.

**The Agreement will page will display information about the repayment plan.**. at top of the page and a check box to agree the plan.

(for example)
**The agreement**	Mrs. Jane Doe will repay £200 in instalments of £50 every month. The first instalment will be paid by 1 January 2023.
**Completion date**	1 May 2023

plus some other static text.

**Review by GorilaNaranja — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Pull request merged**